### PR TITLE
support datasouce closing

### PR DESCRIPTION
**Pull request by smdmts**
https://github.com/scalikejdbc/scalikejdbc/issues/630

### CODE REVIEW

File: scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala [hunk 1]
```diff
@@ -38,5 +54,30 @@ class AuthenticatedDataSourceConnectionPool(
...
+
   override def borrow(): Connection = dataSource.getConnection(user, password)
+
+  override def close(): Unit = closer.close()
+}
+
+/**
+ * Closing support for DataSource
+ */
+trait DataSourceCloser {
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala]: Move this to `DataSourceCloser.scala`.

Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala]: Write some unit tests to show how to use it.

File: scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala [hunk 2]
```diff
@@ -38,5 +54,30 @@ class AuthenticatedDataSourceConnectionPool(
...
+    this(dataSource, user, password, DataSourceConnectionPoolSettings(), closer)
+  }
+
   override def borrow(): Connection = dataSource.getConnection(user, password)
+
+  override def close(): Unit = closer.close()
+}
+
+/**
+ * Closing support for DataSource
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala]: Resource manager which closes a DataSource.

File: scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala [hunk 3]
```diff
@@ -38,5 +54,30 @@ class AuthenticatedDataSourceConnectionPool(
       settings = ConnectionPoolSettings(driverName = settings.driverName)
     ) {
 
+  def this(dataSource: DataSource, user: String, password: String) {
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala]: How about defining like `closer: DataSourceCloser = DefaultDataSourceCloser`? Are these sub-constructors really needed?

Comment (smdmts) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DataSourceConnectionPool.scala]: What I care about break binary compatibility. But if it does not need care that, I will change closer: `DataSourceCloser = DefaultDataSourceCloser` as Default Parameters.

**Pull request merged**